MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java

```diff
@@ -49,25 +57,71 @@ public String getProductPage(ModelMap model){
         return "product";
     }
 
-    @PostMapping("/add")
-    @ResponseStatus(HttpStatus.OK)
-    public void addProduct(@RequestParam String name, @RequestParam String description){
 
-        final String uri = new String(productUri +"/");
+    @GetMapping("/addProduct")
+    public String addProduct(ModelMap model){
+
+        int userId = 1;
+
+        final String unitUri = new String(unitUrl + "/");
+        final String categoryUri = new String(categoryUrl + "{userId}");
+
+        model.addAttribute("product", new Product());
+        //model.addAttribute("image", new Image());
+
+        RestTemplate restTemplate = new RestTemplate();
+
+        unitResult = restTemplate.getForObject(unitUri, Unit[].class);
```

Comment:
Get list right here, instead of converting Array to list below. RestTemplate support this.